plugins/scaffolder-backend/src/scaffolder/tasks/StorageTaskBroker.ts

```diff
@@ -202,6 +205,32 @@ export class StorageTaskBroker implements TaskBroker {
     });
   }
 
+  public async recoverTasks(): Promise<boolean> {
+    const enabled =
+      (this.config &&
+        this.config.getOptionalBoolean(
+          'scaffolder.EXPERIMENTAL_recoverTasks',
+        )) ??
+      false;
+
+    if (enabled) {
+      const recoveredTaskIds =
+        (await this.storage.recoverTasks?.({
+          timeout: readDuration(
+            this.config,
+            'scaffolder.EXPERIMENTAL_recoverTasksTimeout',
+            {
+              seconds: 30,
+            },
+          ),
+        })) ?? [];
+      recoveredTaskIds.forEach(() => {
+        this.signalDispatch();
```

Comment:
Think we only need to do this once, as it's only affecting the current node. The other nodes will eventually run this on their own when there's a new task.